### PR TITLE
Add missing sections and owners to codeowners files

### LINKED ISSUE #1410: Update codeowners file to reflect all areas with approver groups
We have several areas/approver groups  that are not mentioned in the codeowners:
- database 
- messaging 
- feature-flags
- tooling/policies
-

### PR DESCRIPTION
**Pull request by lmolkova**
Fixes #1410

- Add db, messaging, feature-flags and tooling sections to codeowners file
- Add @open-telemetry/semconv-system-approvers as `process` codeowners
- Add @open-telemetry/weaver-maintainers as tooling codeowners
- Make `feature-flag` folder name consistent between model and docs 
 
/also tagging @open-telemetry/semconv-feature-flag-approvers @open-telemetry/semconv-db-approvers @open-telemetry/semconv-messaging-approvers

### CODE REVIEW

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -88,3 +90,21 @@
 /model/deployment/                @open-telemetry/specs-semconv-approvers @open-telemetry/semconv-cicd-approvers
 /model/test/                      @open-telemetry/specs-semconv-approvers @open-telemetry/semconv-cicd-approvers
 /model/vcs/                       @open-telemetry/specs-semconv-approvers @open-telemetry/semconv-cicd-approvers
+
+# Database semantic conventions
+/docs/database/                   @open-telemetry/specs-semconv-approvers @open-telemetry/semconv-db-approvers
```
Comment (lmolkova) [inline .github/CODEOWNERS:95]: note to reviewers: we'll need to give write access to all new teams if this PR is accepted

File: .github/CODEOWNERS [hunk 2]
```diff
@@ -43,9 +43,11 @@
 /model/user-agent/                @open-telemetry/specs-semconv-approvers @open-telemetry/semconv-http-approvers
 
 # System semantic conventions
+/docs/process/                    @open-telemetry/specs-semconv-approvers @open-telemetry/semconv-system-approvers @open-telemetry/semconv-security-approvers
```
Comment (lmolkova) [inline .github/CODEOWNERS:46]: @open-telemetry/semconv-system-approvers I hope you don't mind beling listed as codeowners for `process`. I'm surprised you folks aren't.

**Review by maryliag — APPROVED** [review]
LGTM for the database semantic convention :)

**Review by trask — APPROVED** [review]

**Review by jsuereth — APPROVED** [review]

**Review by mx-psi — APPROVED** [review]

**Review by ChrsMark — APPROVED** [review]

File: .github/CODEOWNERS (hunk 1 above)
Comment (dyladan) [reply .github/CODEOWNERS:94]: Is there a list of the new teams and their members somewhere?

Comment (trask) [reply .github/CODEOWNERS:94]: > Is there a list of the new teams and their members somewhere?

I believe these are all existing teams, we just forgot to add them to the CODEOWNERS file

Comment (lmolkova) [reply .github/CODEOWNERS:94]: added write access to @open-telemetry/semconv-db-approvers @open-telemetry/semconv-messaging-approvers @open-telemetry/weaver-maintainers @open-telemetry/semconv-feature-flag-approvers , so GH now considers this file valid.

**Pull request merged**